src/api/control/member.rs

```diff
@@ -0,0 +1,102 @@
+//! Member definitions and implementations.
+
+use failure::Fail;
+use hashbrown::HashMap;
+
+use crate::log::prelude::*;
+
+/// Error that can be returned by Control API.
+#[derive(Fail, Debug, PartialEq)]
+pub enum ControlError {
+    /// [`Member`] is not found in repository.
+    #[fail(display = "Not found member")]
+    NotFound,
+}
+
+/// ID of [`Member`].
+pub type Id = u64;
+
+/// Media server user with its ID and credentials.
+#[derive(Clone, Debug)]
+pub struct Member {
+    /// ID of [`Member`].
+    pub id: Id,
+    /// Credentials to authorize [`Member`] with.
+    pub credentials: String,
+}
+
+/// Repository that stores store [`Member`]s.
+pub struct MemberRepository {
+    pub members: HashMap<Id, Member>,
+}
+
+impl MemberRepository {
+    /// Returns [`Member`] by its ID.
+    pub fn get_member(&self, id: Id) -> Result<Member, ControlError> {
```

Comment:
@Kirguir no point to apply `Result<Member, ControlError>` here, as `NotFound` is _the only thing_ which can happen here. Even if there was a possibility of some another error, it's still _more idiomatic_ to express optionality with `Option<T>` type and do not bother interface user with errors inspecting: `Result<Option<Member>, RepositoryError>`.

So, just use `Option<Member>`.

Reply:
@tyranron HashMap возвращает `Option<&_>`. Мы же можем и из репозитория возвращать `Option<&Member>?

Reply:
@Kirguir depends what you need do with `Member` after that. `&Member` is cheap, but return _read-only view_ of `Member`. If you want somehow mutate it, then you won't be able to. Also this may complicate concurrent access to `Member`.

You can return owned `Member` by cloning it. The inner `HashMap` should not leak into outer interface and should not form it.